internal/backend/runtime/omni/state_access.go

```diff
@@ -484,6 +495,8 @@ func filterAccessByType(access state.Access) error {
 
 		return status.Error(codes.PermissionDenied, "only read, update and delete access is permitted")
 	case
+		cloud.MachineRequestType,       // read-only for all except for CloudProvider role (checked in filterAccess)
```

Comment:
I guess this comment is valid for the machine request status resource, but it should be possible to manage MachineRequests from Omni, isn't it?
Also, do we apply access policy rules for the MachineRequests?

Reply:
Omni will not be subject to this, as it will be an internal actor - will simply bypass this. (here I assume that `MachineRequest` is not a user-managed resource, therefore users will only be able to read it)

Reply:
About AccessPolicies - didn't do anything related to them, as we don't have the Cluster<>MachineRequest relation yet.

I could actually already add that logic based on a (future) cluster label. But I am not sure if they would carry cluster label - maybe they'll only carry machine class label, don't know yet.

Reply:
gotcha :heavy_check_mark: